source/fe/mapping_q.cc

```diff
@@ -290,12 +290,23 @@ MappingQ<dim, spacedim>::transform_unit_to_real_cell(
   const typename Triangulation<dim, spacedim>::cell_iterator &cell,
   const Point<dim>                                           &p) const
 {
-  return Point<spacedim>(internal::evaluate_tensor_product_value(
-    polynomials_1d,
-    make_const_array_view(this->compute_mapping_support_points(cell)),
-    p,
-    polynomials_1d.size() == 2,
-    renumber_lexicographic_to_hierarchic));
+  if (polynomial_degree > 1)
+    return Point<spacedim>(internal::evaluate_tensor_product_value(
+      polynomials_1d,
+      make_const_array_view(this->compute_mapping_support_points(cell)),
+      p,
+      polynomials_1d.size() == 2,
+      renumber_lexicographic_to_hierarchic));
+  else
+    {
+      const auto vertices = this->get_vertices(cell);
+      return Point<spacedim>(internal::evaluate_tensor_product_value(
```

Comment:
You can use the linear specialization here to avoid checking if linear again https://github.com/dealii/dealii/blob/a8379e2009bdfcb9e6fcb9fdb7dd074668217145/include/deal.II/matrix_free/tensor_product_point_kernels.h#L634